### PR TITLE
feat(service): add helper functions to support static file server

### PR DESCRIPTION
**Pull request by kdada**
**What this PR does / why we need it**:

Support static file. 

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @zxq-bit 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: service/utils.go [hunk 1]
```diff
@@ -39,6 +44,49 @@ func WrapHTTPHandlerFunc(f http.HandlerFunc) func(ctx context.Context) {
...
+
+// UnseekableFile is an error factory to show why can't seek a file.
+// This error may contains private information. Don't return this error to end users directly.
+var UnseekableFile = errors.InternalServerError.Build("Nirvana:Service:UnseekableFile", "can't seek file ${path} because ${reason}")
+
+// ReadFile reads file and returns mime type.
+func ReadFile(path string) (string, io.ReadCloser, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		if os.IsNotExist(err) || os.IsPermission(err) {
```
Comment (zxq-bit) [inline service/utils.go]: `os.IsPermission(err)` 用 403 可能更好些？

Comment (kdada) [reply service/utils.go]: Right.

**Pull request merged**